### PR TITLE
config: Azure Stack Hub support

### PR DESCRIPTION
**Pull request by patrickdillon**
Add Azure Stack Hub as a valid Azure cloud environment.

See openshift/enhancements#689

cc @staebler

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ type AzurePlatformStatus struct {
 }
 
 // AzureCloudEnvironment is the name of the Azure cloud environment
-// +kubebuilder:validation:Enum="";AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
+// +kubebuilder:validation:Enum="";AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud;AzureStackCloud;
```
Comment (staebler) [inline config/v1/types_infrastructure.go:340]: Is it valid for this to end in a `;`?

Comment (patrickdillon) [reply config/v1/types_infrastructure.go]: Nice catch. I don't think so, so I fixed it, despite the passing tests, generated docs.

**Comment by deads2k** [comment]
I'm ok adding this, but the enhancement should be agreed and merged before merging the API.

**Comment by staebler** [comment]
> I'm ok adding this, but the enhancement should be agreed and merged before merging the API.

@deads2k The enhancement has merged.

**Comment by patrickdillon** [comment]
@deads2k would you mind taking another look at this?

/assign @deads2k

**Comment by patrickdillon** [comment]
Lost the `/lgtm` rebasing. @staebler can you reapply?

**Comment by patrickdillon** [comment]
I have added a field for the ARM Endpoint in 6734eef7cb10c381a42893aeb2ebf729f77513dc This is redundant information that is already captured in the cloud provider config, but the infrastructure object may be a preferable place to discover the endpoints.

If this API field is approved, we will update the cloud config operator to monitor this field, and upon a change, update the cloud provider config.

**Pull request merged**